### PR TITLE
NF-159 Require x-correlation-id header to be provided

### PR DESCRIPTION
**Pull request by crosslandwa**
Update EIS create-case connector to specify x-request-id (instead of passing _all_ MDTP headers)

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+trait WithCorrelationId { self: Results =>
+
+  protected def withCorrelationId(
+    f: String => Future[Result]
+  )(implicit ec: ExecutionContext, request: Request[_]): Future[Result] =
+    request.headers.get("x-correlation-id") match {
+      case Some(value) => f(value)
+      case None =>
+        Future.successful(
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala:33]: Just `Future`?

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/CreateCaseConnector.scala [hunk 2]
```diff
@@ -43,7 +43,10 @@ class CreateCaseConnector @Inject() (val config: AppConfig, val http: HttpPost)(
     http.POST[EISCreateCaseRequest, EISCreateCaseResponse](url, request)(
       implicitly[Writes[EISCreateCaseRequest]],
       readFromJsonSuccessOrFailure,
-      hc.copy(authorization = Some(Authorization(s"Bearer ${config.eisAuthorizationToken}")))
+      HeaderCarrier(
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/CreateCaseConnector.scala:46]: Just for my understanding.   We are deliberately not passing this new `x-correlation-id` onto EIS (or any other curated MDTP headers)?   That bearer token that we'll be sending them is the only thing that matters in terms of getting access to EIS?

**Review by SteveSugden — DISMISSED** [review]
LGTM

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/CreateCaseConnector.scala (hunk 2 above)
Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/CreateCaseConnector.scala:46]: Theres a `.withExtraHeaders(pegaApiHeaders(...` call at line 50 that adds the `x-correlation-id` and other EIS specific headers - arguably the `authorization` header should be added in there too (but that would require the `PegaConnector` trait to have access to `AppConfig`)

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala (hunk 1 above)
Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala]: I'll pre-compute the (static) response and return it via `Future.successful`

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/CreateCaseConnector.scala (hunk 2 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/connectors/CreateCaseConnector.scala:46]: :+1 Ah, missed that bit

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      "message"    -> JsString("Missing header x-correlation-id")
+    )
+  )
+
+  protected def withCorrelationId(
+    f: String => Future[Result]
+  )(implicit ec: ExecutionContext, request: Request[_]): Future[Result] =
+    request.headers.get("x-correlation-id") match {
+      case Some(value) => f(value)
+      case None        => Future.successful(missingXCorrelationIdResponse)
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala:39]: Why not `Future(missingXCorrelationIdResponse)` ?

Comment (crosslandwa) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala:39]: The sync code in `Future.successful` returns a pre-computed value - there is no _work_ to be done synchronously, so the call does not block

Doing `Future(precomputedResult)` would incur overheads for setting up/executing the async execution of `Future.apply`

Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentre/controllers/WithCorrelationId.scala:39]: That makes sense.

**Review by SteveSugden — APPROVED** [review]

**Pull request merged**